ecpy_pulses/sequences/views/sequence_editor_view.enaml

```diff
@@ -0,0 +1,492 @@
+                    if self.operations:
+                        button.show()
+                    view.show()
+
+            elif change.removed:
+                # Remove all of the children that were deleted:
+                for _, item in change.removed:
+                    self._remove_view(item)
+
+            elif change.moved and root:
+                # Simply need to relayout.
+                self.request_relayout()
+
+            # Make sure the _empty_button is correctly parented.
+            if self._children_buttons:
+                self._empty_button.set_parent(None)
+            elif 'addP' in self.operations:
+                self._empty_button.set_parent(self)
+                self._empty_button.show()
+
+        else:
+            # The whole member was replaced so we can just forget about its
+            # previous state. This should be a rare event.
+
+            # First we destroy all the buttons and views (not worth optimizing
+            # more).
+            for b, w in self._children_buttons.values():
+                b.destroy()
+                w.destroy()
+
+            self._empty_button.set_parent(None)
+
+            # Next we repopulate the container.
+            self._populate()
+
+    def _remove_view(self, item):
+        """Remove the view associated with a task.
+
+        """
+        button, view = self._children_buttons.pop(item)
+        button.destroy()
+        view.destroy()
+
+    def _default_operations(self):
+        """Build the operations dictionary used by the popup menu.
+
+        """
+        def addP(index, position):
+            """Handler for the add entry of the menu.
+
+            """
+            pulse = Pulse()
+            if pulse:
+                if position == 'after':
+                    index += 1
+                self.sequence.add_child_item(index, pulse)
+
+        def addS(index, position):
+            """Handler for the add entry of the menu.
+
+            """
+            obj = self.core.invoke_command('ecpy.pulses.build_sequence',
+                                      {'parent_ui': self,
+                                       'mode': 'ui',
+                                       'root': self.sequence.root}, self)
+            if obj:
+                if obj:
+                    if position == 'after':
+                        index += 1
+                    self.sequence.add_child_item(index, obj)
+
+        def move(old, new):
+            """Handler for the move entries of the menu.
+
+            """
+            self.sequence.move_child_item(old, new)
+
+        def remove(index):
+            """Handler for the remove entry of the menu.
+
+            """
+            self.sequence.remove_child_item(index)
+
+        return {'addP': addP, 'addS': addS,
+                'move': move, 'remove': remove}
+
+    def _default__empty_button(self):
+        """Button used to add the fist task.
+        """
+        return EmptyEditorButtons(editor=self)
+
```

Comment:
Ideally this would live on the RootSequenceView as everybody has a reference to it